### PR TITLE
feat(confluence): add installation & users sync & data protection

### PR DESCRIPTION
**Pull request by elbalexandre**
## Description

- add a type in `@elba-security/schemas` (infered from a schema)
- update inngest test util
- add complete confluence integration

## Additional Notes

N/A

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/confluence/src/connectors/confluence/auth.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      client_secret: env.CONFLUENCE_CLIENT_SECRET,
+      refresh_token: refreshToken,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+
+  if (!response.ok) {
+    throw new ConfluenceError('Could not retrieve token', { response });
```
Comment (elbalexandre) [inline apps/confluence/src/connectors/confluence/auth.ts:61]:
```suggestion
    throw new ConfluenceError('Could not refresh token', { response });
```

File: apps/confluence/src/inngest/client.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        groupId: string;
+        cursor: number | null;
+      };
+    };
+    'confluence/group.users.sync.completed': {
+      data: {
+        organisationId: string;
+        groupId: string;
+      };
+    };
```
Comment (elbalexandre) [inline apps/confluence/src/inngest/client.ts:91]: is it used ?

File: apps/confluence/src/database/schema.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      .notNull()
+      .references(() => organisationsTable.id, { onDelete: 'cascade' }),
+    lastSyncAt: timestamp('last_sync_at').notNull(),
+    publicName: text('public_name').notNull(),
+    displayName: text('display_name'),
+    createdAt: timestamp('created_at').defaultNow().notNull(),
+  },
+  (t) => ({
+    // using lastSyncAt allows duplicate data during updates
+    unq: unique().on(t.organisationId, t.id, t.lastSyncAt),
```
Comment (proquig) [inline apps/confluence/src/database/schema.ts:27]: I'm not sure to understand the purpose of this? Why not just make orgId - userId unique and update existing users lastSyncAt on conflict?

File: apps/confluence/src/inngest/functions/users/sync-group-users.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    const nextCursor = await step.run('paginate-group-users', async () => {
+      const elba = createElbaClient(organisation.id, organisation.region);
+      const result = await getGroupMembers({
+        accessToken: await decrypt(organisation.accessToken),
+        instanceId: organisation.instanceId,
+        groupId,
+        cursor,
+      });
+      const altassianMembers = result.members.filter((user) => user.accountType === 'atlassian');
```
Comment (proquig) [inline apps/confluence/src/inngest/functions/users/sync-group-users.ts:39]:
```suggestion
      const atlassianMembers = result.members.filter((user) => user.accountType === 'atlassian');
```

**Review by proquig — APPROVED** [review]

File: apps/confluence/src/database/schema.ts (hunk 3 above)
Comment (elbalexandre) [reply apps/confluence/src/database/schema.ts:27]: Let me check, I guess it's useless.

**Pull request merged**